pkg/dynamicinstrumentation/codegen/templates.go

```diff
@@ -0,0 +1,202 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package codegen
+
+var forcedVerifierErrorTemplate = `
+int illegalDereference = *(*(*ctx->regs[0]));
+`
+
+var headerTemplateText = `
+// Name={{.Name}} ID={{.ID}} TotalSize={{.TotalSize}} Kind={{.Kind}}
+// Write the kind and size to output buffer
+param_type = {{.Kind}};
+bpf_probe_read_kernel(&event->output[outputOffset], 1, &param_type);
+param_size = {{.TotalSize}};
+bpf_probe_read_kernel(&event->output[outputOffset+1], 2, &param_size);
+outputOffset += 3;
+`
+
+// The length of slices aren't known until parsing, so they require
+// special headers to read in the length dynamically
+var sliceRegisterHeaderTemplateText = `
+// Name={{.Parameter.Name}} ID={{.Parameter.ID}} TotalSize={{.Parameter.TotalSize}} Kind={{.Parameter.Kind}}
+// Write the slice kind to output buffer
+param_type = {{.Parameter.Kind}};
+bpf_probe_read_kernel(&event->output[outputOffset], 1, &param_type);
+// Read slice length and write it to output buffer
+bpf_probe_read_user(&param_size, 8, &ctx->regs[{{.Parameter.Location.Register}}+1]);
+bpf_probe_read_kernel(&event->output[outputOffset+1], 2, &param_size);
+outputOffset += 3;
```

Comment:
If `outputOffset` goes beyond `ArgumentsMaxSize` the you would have a verification failure. Should this be checked in the eBPF side?